lang/java/reef-utils/src/main/java/org/apache/reef/util/MultiAsyncToSync.java

```diff
@@ -82,11 +94,15 @@ public MultiAsyncToSync(final long timeoutPeriod, final TimeUnit timeoutUnits) {
       addSleeper(identifier, call);
       // Invoke the caller's asynchronous processing while holding the lock
       // so a wakeup cannot occur before the caller sleeps.
-      asyncProcessor.run();
+      if (executor == null) {
+        asyncProcessor.run();
+      } else {
+        executor.execute(asyncProcessor);
+      }
       // Put the caller to sleep until the ack comes back. Note: we atomically
       // give up the look as the caller sleeps and atomically reacquire the
       // the lock as we wake up.
-      LOG.log(Level.FINER, "Putting caller to sleep on identifier [{0}]", identifier);
+      LOG.log(Level.INFO, "Putting caller to sleep on identifier [{0}]", identifier);
```

Comment:
I don't like that logging level. at least put some FIXME comments for each so we can change it back later. better yet, figure out how to set log level properly in Java (answer: set 
`-Djava.util.logging.config.class=org.apache.reef.util.logging.Config`
in java command line)

Reply:
Okay, I sent a message to the dev list but do not have a reply.

Reply:
Added FIXME